tests/datasets/test_mapinwild.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import builtins
+import glob
+import os
+import shutil
+from pathlib import Path
+from typing import Any
+
+import matplotlib.pyplot as plt
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from _pytest.monkeypatch import MonkeyPatch
+from torch.utils.data import ConcatDataset
+
+import torchgeo.datasets.utils
+from torchgeo.datasets import MapInWild
+
+pytest.importorskip("pandas", minversion="1.1.3")
+
+
+def download_url(url: str, root: str, *args: str, **kwargs: str) -> None:
+    shutil.copy(url, root)
+
+
+class TestMapInWild:
+    @pytest.fixture(params=["train", "validation", "test"])
+    def dataset(self, tmp_path: Path, monkeypatch: MonkeyPatch) -> MapInWild:
+        monkeypatch.setattr(torchgeo.datasets.mapinwild, "download_url", download_url)
+
+        md5s = {
+            "ESA_WC.zip": "b14b6b54ef33ffd6e5802fb3096b0710",
+            "VIIRS.zip": "012734f798a39fdddf24fd9bdfc5f790",
+            "mask.zip": "d7fcb0ac6fc51a0189c60a8c44403194",
+            "s1_part1.zip": "17a9e23df5063d066098c84a4074cef1",
+            "s1_part2.zip": "c6748e547f18288300d670e70bff234a",
+            "s2_autumn_part1.zip": "89890200ef79bd80c6cbf1d4bbd314c6",
+            "s2_autumn_part2.zip": "22b7f6be3e59daaf264a179c94498e8e",
+            "s2_spring_part1.zip": "5f91cd21d3929b1e384b128b90f4efe3",
+            "s2_spring_part2.zip": "5ab77f7352d42cc5de76bf7293e20877",
+            "s2_summer_part1.zip": "d80b22a94c5fa0aca61fdc9d8cc7c361",
+            "s2_summer_part2.zip": "2ec2f215fcc8247df64f50288f664a8b",
+            "s2_temporal_subset_part1.zip": "0398bc292d9c3e8513444c96c5b0da96",
+            "s2_temporal_subset_part2.zip": "c063386289476dc2342872abb7c57f1c",
+            "s2_winter_part1.zip": "4f67835837f865f03cda9841b0da0717",
+            "s2_winter_part2.zip": "b6654fb29a315bddaca8b3e7a9f34930",
+        }
+        monkeypatch.setattr(MapInWild, "md5s", md5s)
+
+        urls = os.path.join("tests", "data", "mapinwild")
+
+        modality_urls = {
```

Comment:
You can now replace this entire thing with a one-line monkeypatch of `MapInWild.url`

Reply:
I did not quite get this one. I monkeypatch the attribute "modality_url" below and there is no "MapInWild.url" attribute. Could you help me out on this?

Reply:
There is a `MapInWild.url` attribute, it's value is:
```python
url = "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/"
```

Reply:
You might need to add a subdirectory to your tarballs to match.